### PR TITLE
HDCP Task #228 Submission

### PR DESCRIPTION
**Pull request by deoidh**
It was fun, learnt a few things along the way I'm happy to say here is my submission!

[Contract source](https://github.com/samgos/hydrogen-oxide)
[Rinkeby deployment
](https://rinkeby.etherscan.io/address/0x070a6ea824d87780643e888196cd92d748b02a22)

I hope to get on the other HYDRO tasks available, thanks to the team for being compliant and creating a great community development program 🎉

### CODE REVIEW

**Comment by NoahHydro** [comment]
Awesome @samgos ! Mind fixing all the lint errors and pushing a new commit before I review? You can see them in the Travis logs, or locally with `npm run lint`.

**Comment by samgos** [comment]
@NoahHydro hope it suffices?

**Comment by NoahHydro** [comment]
@samgos can you verify the source code on etherscan? we should have it more clear that this is a requirement, my fault!

https://rinkeby.etherscan.io/address/0xD9D40Ca7BedcC47E5D4084c4011dAc50566F8d4f

**Comment by samgos** [comment]
New deployment after testing on Kovan and Rinkeby, to ensure the Oraclize responses were recurring and verified contract source code at: https://rinkeby.etherscan.io/address/0x2930cf9ee8e03c3e06fa1828ccd8e371323fde0f#code

File: src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+  const [oxideBalance, setOxide]  = useState(0)
+  const [committedWager, setWager]  = useState(0)
+  const [activePunters, setPunters]  = useState(0)
+  const [open, setOpen] = useState(false);
+  const [leaderboardData, setLeaderboard]  = useState([])
+  const oxideContract = useGenericContract('0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f', ABI)
+  const snowflakeBalance = useSnowflakeBalance(ein)
+
+  function refreshLeaderboard(_round)  {
+    oxideContract.getPastEvents("scoreLog", { fromBlock: 0, toBlock: 'latest' })
```
Comment (NoahHydro) [inline src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js]:
```suggestion
    oxideContract.getPastEvents("scoreLog", { fromBlock: 3795013, toBlock: 'latest' })
```

Comment (NoahHydro) [reply src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js]: This is the block height that snowflake was deployed at, and will speed up the event fetching stage.

File: src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+              *parseObject(result[x].returnValues.wagerAmount),
+            )))
+          }
+        }
+      setLeaderboard(leaderboard)
+    })
+  }
+
+  function getWinner(_round){
+    oxideContract.getPastEvents("winnerAlert", { fromBlock: 0, toBlock: 'latest' })
```
Comment (NoahHydro) [inline src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js]:
```suggestion
    oxideContract.getPastEvents("winnerAlert", { fromBlock: 3795013, toBlock: 'latest' })
```

File: src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        _round = parseInt(parseObject(_round))
+        if(round === (_round-1)){
+          setWinner(parseInt(parseObject(result[x].returnValues.wagerEIN)))
+        }
+      }
+    })
+  }
+
+  function watchWinner() {
+    oxideContract.events.winnerAlert({ fromBlock: 0 },
```
Comment (NoahHydro) [inline src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js]:
```suggestion
    oxideContract.events.winnerAlert({ fromBlock: 3795013 },
```

File: src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+    .on('data', (event) => {
+      var round = parseInt(parseObject(event.returnValues.wagerRound))
+      if(round === (activeRound - 1)){
+        setWinner(event.returnValues.wagerEIN)
+      }
+    })
+  }
+
+  function watchScores(_round){
+    oxideContract.events.scoreLog({ fromBlock: 0 },
```
Comment (NoahHydro) [inline src/components/resolvers/Rinkeby/0x2930Cf9EE8E03C3E06Fa1828cCD8E371323Fde0f/Oxide.js]:
```suggestion
    oxideContract.events.scoreLog({ fromBlock: 3795013 },
```

**Review by ghost — APPROVED** [review]

**Comment by NoahHydro** [comment]
great work @samgos! once you accept my code review suggestions, i'll merge this and it'll be live. @AnuragHydro will follow up with a) information on when we'll send you your bounty, and b) some suggestions for UX improvements if you're ever so motivated in the future.

**Comment by samgos** [comment]
@NoahHydro  resolved!

**Pull request merged**